src/main/java/io/appform/jsonrules/expressions/preoperation/date/DiffFromCurrentEpochOperation.java

```diff
@@ -0,0 +1,49 @@
+package io.appform.jsonrules.expressions.preoperation.date;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.appform.jsonrules.ExpressionEvaluationContext;
+import io.appform.jsonrules.expressions.preoperation.PreOperation;
+import io.appform.jsonrules.expressions.preoperation.PreOperationType;
+import io.appform.jsonrules.utils.PreOperationUtils;
+import lombok.Builder;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+
+import java.time.Instant;
+import java.time.OffsetDateTime;
+
+/**
+ * Created by shubham.ppe on 24/07/17.
+ */
+@Data
+@Builder
+@EqualsAndHashCode(callSuper = true)
+@ToString(callSuper = true)
+public class DiffFromCurrentEpochOperation extends PreOperation<Number> {
+    protected DiffFromCurrentEpochOperation() {
+        super(PreOperationType.current_epoch_diff);
+    }
+
+
+    private Number compute(JsonNode evaluatedNode, long currentEpoch) {
+        if (evaluatedNode.isIntegralNumber()) {
+            return currentEpoch - evaluatedNode.asLong();
+        }
+        throw new IllegalArgumentException("Evaluated node does not represent a valid epoch");
+    }
+
+    @Override
+    public Number compute(ExpressionEvaluationContext context) {
+        try {
+            long currentEpoch = getCurrentTime(context);
+            return compute(context.getNode(), currentEpoch);
+        } catch (Exception e) {
+            throw new IllegalArgumentException("Operands does not represent a valid epoch", e);
+        }
+    }
+
+    public long getCurrentTime(ExpressionEvaluationContext context) {
+        return (long)(context.getOptions().getOrDefault(ExpressionEvaluationContext.OPTIONS_KEYS.SYSTEM_TIME.name(), Instant.now().getEpochSecond()));
```

Comment:
Second? Should be millisencond .. no? Also, why not System.currentTimeMillis()? There is no reuse for this function. Seems like an overkill to make a function out of it.